legion/operator/pkg/utils/docker.go

```diff
@@ -0,0 +1,116 @@
+//
+//    Copyright 2019 EPAM Systems
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/docker/distribution/manifest/schema1"
+	"github.com/heroku/docker-registry-client/registry"
+	"github.com/legion-platform/legion/legion/operator/pkg/legion"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+)
+
+var (
+	dockerImageRegexp = regexp.MustCompile(`(.*?)/([\w\-/]+):([\-.\w]+)`)
```

Comment:
This regexp does not cover root images (`alpine` and etc.). I'm not sure that this is critical, but it should be documented.

Reply:
Fixed